api/src/main/java/edu/cornell/mannlib/vitro/webapp/search/controller/PagedSearchController.java

```diff
@@ -151,49 +153,64 @@ protected ResponseValues processRequest(VitroRequest vreq) {
+            if (log.isDebugEnabled()) {
+                log.debug(getSpentTime(startTime) + "ms spent before read filter configurations.");
+            }
+            Map<String, SearchFilter> filterConfigurationsByField = SearchFiltering.readFilterConfigurations(vreq);
+            if (log.isDebugEnabled()) {
+                log.debug(getSpentTime(startTime) + "ms spent before get sort configurations.");
+            }
+            Map<String, SortConfiguration> sortConfigurations = SearchFiltering.getSortConfigurations(vreq);
+            if (log.isDebugEnabled()) {
+                log.debug(getSpentTime(startTime) + "ms spent before get query configurations.");
+            }
+            SearchQuery query =
+                    getQuery(queryText, hitsPerPage, startIndex, vreq, filterConfigurationsByField, sortConfigurations);
+            if (log.isDebugEnabled()) {
+                log.debug(getSpentTime(startTime) + "ms spent after get query configurations.");
             }
-
-            SearchQuery query = getQuery(queryText, hitsPerPage, startIndex, vreq);
             SearchEngine search = ApplicationUtils.instance().getSearchEngine();
             SearchResponse response = null;
 
             try {
                 response = search.query(query);
             } catch (Exception ex) {
                 String msg = makeBadSearchMessage(queryText, ex.getMessage(), vreq);
-                log.error("could not run search query",ex);
+                log.error("could not run search query", ex);
                 return doFailedSearch(msg, queryText, format, vreq);
             }
+            if (log.isDebugEnabled()) {
+                log.debug(getSpentTime(startTime) + "ms spent after get query execution.");
+            }
 
             if (response == null) {
                 log.error("Search response was null");
                 return doFailedSearch(I18n.text(vreq, "error_in_search_request"), queryText, format, vreq);
             }
-
+            addFacetCountersFromRequest(response, filterConfigurationsByField, vreq);
+            if (log.isDebugEnabled()) {
+                log.debug(getSpentTime(startTime) + "ms spent after addFacetCountersFromRequest.");
+            }
             SearchResultDocumentList docs = response.getResults();
             if (docs == null) {
                 log.error("Document list for a search was null");
-                return doFailedSearch(I18n.text(vreq, "error_in_search_request"), queryText,format, vreq);
+                return doFailedSearch(I18n.text(vreq, "error_in_search_request"), queryText, format, vreq);
             }
 
             long hitCount = docs.getNumFound();
             log.debug("Number of hits = " + hitCount);
-            if ( hitCount < 1 ) {
-                return doNoHits(queryText,format, vreq);
-            }
+            // if ( hitCount < 1 ) {
+            // return doNoHits(queryText,format, vreq);
+            // }
```

Comment:
I would remove those three lines. The doNoHits method is already removed

Reply:
removed